### ISSUE TITLE
Infer Log Analytics Monitoring Costs

### ISSUE THREAD

**Issue by Gordonby** [TheCloudTheory/arm-estimator#108]
I notice that the Log Analytics Calculation could be made a little more sophisticated.

2 ideas.

1. Use the LogDataCap property as an absolute estimate. If 1GB is capped, then estimate based on the cap.
2. Use metadata to provide hints to feed the estimation calculation

This arm json file was generated using [metadata](https://github.com/Azure/bicep/issues/6126) to hint and the datacap mentioned.
```json
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "metadata": {
    "_generator": {
      "name": "bicep",
      "version": "0.12.40.16777",
      "templateHash": "14274997296602342882"
    },
    "estimateHints": {
      "Microsoft.OperationalInsights/workspaces": "1gb"
    }
  },
  "parameters": {
    "location": {
      "type": "string",
      "defaultValue": "[resourceGroup().location]"
    },
    "logDataCap": {
      "type": "int",
      "defaultValue": 1
    }
  },
  "resources": [
    {
      "type": "Microsoft.OperationalInsights/workspaces",
      "apiVersion": "2022-10-01",
      "name": "log",
      "location": "[parameters('location')]",
      "properties": "[union(createObject('retentionInDays', 30, 'sku', createObject('name', 'PerGB2018')), if(greater(parameters('logDataCap'), 0), createObject('workspaceCapping', createObject('dailyQuotaGb', parameters('logDataCap'))), createObject()))]",
      "metadata": {
        "description": "The logging resource"
      }
    }
  ]
}
```

**Comment by kamil-mrzyglod** [comment]
Not sure I follow this proposal :)

You mean ACE could base its estimations for Log Analytics on hints provided in the `metadata`? If so, this seems connected to #16

**Comment by Gordonby** [comment]
> Not sure I follow this proposal :)

Log Analytics SKU's generally charge by the GB.
In order to create a meaningful estimate for a new resource, you'll need some indication of the expected daily/monthly usage.

I was proposing 2 mechanisms for creating a meaningful estimate.


> You mean ACE could base its estimations for Log Analytics on hints provided in the metadata? If so, this seems connected to https://github.com/TheCloudTheory/arm-estimator/issues/16

Yes, that's what I'm saying. Metadata is just 1 option.

**Comment by Gordonby** [comment]
The same mechanism would work for other services like ACR

![image](https://user-images.githubusercontent.com/17914476/204023670-0e964995-462f-4f1f-9d6b-36d86ed15129.png)

**Comment by kamil-mrzyglod** [comment]
Yeah, so that's exactly what's planned for `1.0` release. That's a general challenge for better estimates as there're lots of resources, which rely on providing some expected usage of metrics, which cannot be easily estimated. 

For now I've been thinking about a configuration file (JSON / YAML), which can be optionally provided and used as input for estimations. BUT I must say use of a native construct from Bicep seems very tempting. I'll definitely give it a try :)

**Comment by kamil-mrzyglod** [comment]
Preview of usage patterns (for ACR) is available with `1.0` release, moving Log Analytics to `1.1` as part of further work in #16
